inspire_json_merger/utils/filterout_utils.py

```diff
@@ -59,24 +64,67 @@ def filter_conflicts_by_path(conflict_list, to_delete_path):
 
 
 def is_to_delete(conflict, keys_path):
-    to_delete = keys_path.split('.')
-    conflict_path = conflict_to_list(conflict)
+    conflict_path = '.'.join(conflict_to_list(conflict))
+    return conflict_path.startswith(keys_path)
 
-    if len(to_delete) > len(conflict_path):
-        return False
 
-    i = 0
-    while i < len(conflict_path) and i < len(to_delete):
-        c = conflict_path[i]
-        d = to_delete[i]
+def conflict_to_list(conflict):
+    path = conflict[1]
+    return [p for p in path if not isinstance(p, int)]
 
-        if c != d:
-            return False
-        i += 1
 
-    return True
+def _remove_source_from_field(source, field):
+    """Remove all elements matching ``source`` in ``field``."""
+    return [element for element in field if element.get('source') != source]
 
 
-def conflict_to_list(conflict):
-    path = conflict[1]
-    return [p for p in path if not isinstance(p, int)]
+def keep_only_update_source_in_field(field, root, head, update):
+    """Remove elements from root and head where ``source`` matches the update.
+
+    This is useful if the update needs to overwrite all elements with the same
+    source.
+
+    .. note::
+        If the update doesn't contain exactly one source in ``field``, the
+        records are returned with no modifications.
+
+    Args:
+        field(str): the field to filter out.
+        root(dict): the root record, whose ``field`` will be cleaned.
+        head(dict): the head record, whose ``field`` will be cleaned.
+        update(dict): the update record, from which the ``source`` is read.
+
+    Returns:
+        tuple: ``(root, head, update)`` with some elements filtered out from
+            ``root`` and ``head``.
+    """
+    update_sources = set(get_value(update, '.'.join([field, 'source']), []))
+    if len(update_sources) != 1:
```

Comment:
I know the update it should not have >1 source, but well, maybe better throw a warning if it does instead of just going on. Hmm, maybe a desy harvest might have more than one source...

Reply:
yes, and in that case we are not dealing with a regular update, so we skip the erasing and fall back to the regular strategies with the comparators (that I did not remove).